addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/NetworkOceanicThingHandler.java

```diff
@@ -0,0 +1,222 @@
+                                    if (tmpData[i - 2] == eChar && tmpData[i - 1] == rChar && tmpData[i] == rChar) {
+                                        // Received ERR from the device.
+                                        return null;
+                                    }
+                                }
+
+                                if (sequenceFound && i > 0
+                                        && (tmpData[i - 1] != carriageReturn && tmpData[i] == nullChar)) {
+                                    index = 0;
+                                    // Ignore trash received
+                                    if (logger.isTraceEnabled()) {
+                                        StringBuilder sb = new StringBuilder();
+                                        for (int j = 0; j < i; j++) {
+                                            sb.append(String.format("%02X", tmpData[j]));
+                                            sb.append(" ");
+                                        }
+                                        logger.trace("Ingoring {} bytes : {}", i, sb.toString());
+                                    }
+                                }
+
+                                if (sequenceFound && (tmpData[i] == carriageReturn)) {
+                                    if (index > 0) {
+                                        line = new String(Arrays.copyOf(dataBuffer, index));
+                                        logger.debug("Received response '{}'",
+                                                new String(Arrays.copyOf(dataBuffer, index)));
+                                        line = StringUtils.chomp(line);
+                                        line = line.replace(",", ".");
+                                        line = line.trim();
+                                        index = 0;
+
+                                        return line;
+                                    }
+                                }
+
+                                if (index == bufferSize) {
+                                    index = 0;
+                                }
+                            }
+                        }
+
+                    }
+                } catch (Exception e) {
+                    logger.error("An exception ocurred while quering host {}:{} : '{}'", config.ipAddress,
+                            config.portNumber, e.getMessage(), e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    reconnectJob = scheduler.schedule(reconnectRunnable, RECONNECT_INTERVAL, TimeUnit.SECONDS);
+                } finally {
+                    Throttler.unlock(config.ipAddress);
+                }
+            }
+            return null;
+        }
+    }
+
+    private Runnable reconnectRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            dispose();
```

Comment:
Hmm, should we have this?

Reply:
Was the easiest way to close both the socket and kill the polling job